components/launcher/src/server/mod.rs

```diff
@@ -403,18 +387,31 @@ where
 }
 
 pub fn run(args: Vec<String>) -> Result<i32> {
-    let mut server = Server::new(args)?;
+    let mut restart = false;
+    let restart_sleep_secs = core::env::var(SUP_RESTART_SLEEP_SECS)
+        .unwrap_or("".to_string())
+        .parse()
+        .unwrap_or(DEFAULT_SUP_RESTART_SLEEP_SECS);
+
     signals::init();
     loop {
-        match server.tick() {
-            Ok(TickState::Continue) => thread::sleep(Duration::from_millis(100)),
-            Ok(TickState::Exit(code)) => {
-                return Ok(code);
+        let mut server = Server::new(&args, restart)?;
+        restart = true;
+        loop {
+            match server.tick() {
+                Ok(TickState::Continue) => thread::sleep(Duration::from_millis(100)),
+                Ok(TickState::Exit(code)) => return Ok(code),
+                Err(e) => {
+                    debug!("tick() returned error {}. Attempting to restart server", e);
+                    break;
+                }
             }
-            Err(_) => while server.reload().is_err() {
-                thread::sleep(Duration::from_millis(1_000));
-            },
         }
+        error!(
+            "Supervised exited; will restart in {} secs...",
```

Comment:
"Supervised" => "Supervisor"

Reply:
Could also say "seconds", too.

Reply:
Ooh good catch on the typo, will fix.

But why the "secs negativity"? 😜